packages/synapse-react-client/src/components/SynapseTable/datasets/DatasetItemsEditor.tsx

```diff
@@ -253,6 +313,36 @@ export function DatasetItemsEditor(props: DatasetItemsEditorProps) {
       },
     })
 
+  useEffect(() => {
+    // When changes are made (but not saved) to the datasetToUpdate, show a notification that summarizes the change and prompts the user to save
+    if (
+      previousDatasetToUpdate &&
+      datasetToUpdate &&
+      !isEqual(previousDatasetToUpdate.items, datasetToUpdate.items) &&
+      !updateIsPending
```

Comment:
When deciding whether to push a toast to prompt the user to commit their unsaved changes,
1. Only compare `items`, instead of the entire entity
2. Don't push a toast if the update call is in flight